src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -309,15 +312,15 @@ private static AnalyzerConfigDocument CreateAnalyzerConfigDocument(DocumentId do
         /// cases you should call <see cref="GetCompilationAsync"/> which will either return the cached <see cref="Compilation"/>
         /// or create a new one otherwise.
         /// </summary>
-        public bool TryGetCompilation(out Compilation compilation)
+        public bool TryGetCompilation([NotNullWhen(returnValue: true)] out Compilation? compilation)
         {
             return _solution.State.TryGetCompilation(this.Id, out compilation);
         }
 
         /// <summary>
         /// Get the <see cref="Compilation"/> for this project asynchronously.
         /// </summary>
-        public Task<Compilation> GetCompilationAsync(CancellationToken cancellationToken = default)
+        public Task<Compilation?> GetCompilationAsync(CancellationToken cancellationToken = default)
```

Comment:
❓ Why does this allow null returns?

Reply:
It'll return null for F#/TypeScript projects. This is actually a huge source of bugs we've had historically where people forget to deal with this.

Reply:
💡 Consider updating the documentation comment to clarify.

Reply:
Good idea.